cli_tests/app.sh

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env bats
+
+# To run locally:
+# export KETCH_COMMAND=<location of ketch binary>
+# assure you have a kubernetes cluster running w/ traefik, cert manager, etc. (see ketch getting started docs)
+# assure the ketch cli is compiled (make ketch)
+# assure you have bats installed locally (via apt, brew, etc.)
+# ./cli_tests/app.sh
+
+setup() {
+    if [[ -z "${KETCH_COMMAND}" ]]; then
+    KETCH=$(pwd)/bin/ketch
+  else
+    KETCH="${KETCH_COMMAND}"
+  fi
+  INGRESS=$(kubectl get svc traefik -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+  FRAMEWORK="myframework"
+  APP_IMAGE="docker.io/shipasoftware/bulletinboard:1.0"
+  APP_NAME="bulletinboard"
+  CNAME="my-cname.com"
+  TEST_ENVVAR_KEY="FOO"
+  TEST_ENVVAR_VALUE="BAR"
+}
+
+@test "help" {
+  result="$($KETCH help)"
+  echo "RECEIVED:" $result
+  [[ $result =~ "For details see https://theketch.io" ]]
+  [[ $result =~ "Available Commands" ]]
+  [[ $result =~ "Flags" ]]
+}
+
+@test "framework add" {
+  result=$($KETCH framework add "$FRAMEWORK" --ingress-service-endpoint "$INGRESS" --ingress-type "traefik")
+  echo "RECEIVED:" $result
+  [[ $result =~ "Successfully added!" ]]
+}
+
+@test "framework list" {
+  result=$($KETCH framework list)
+  headerRegex="NAME[ \t]+STATUS[ \t]+NAMESPACE[ \t]+INGRESS TYPE[ \t]+INGRESS CLASS NAME[ \t]+CLUSTER ISSUER[ \t]+APPS"
+  dataRegex="$FRAMEWORK[ \t]+ketch-$FRAMEWORK[ \t]+traefik[ \t]+traefik"
+  echo "RECEIVED:" $result
+  [[ $result =~ $headerRegex ]]
+  [[ $result =~ $dataRegex ]]
+}
+
+@test "framework update" {
+  result=$($KETCH framework update "$FRAMEWORK" --app-quota-limit 1)
+  echo "RECEIVED:" $result
+  [[ $result =~ "Successfully updated!" ]]
+}
+
+@test "app deploy" {
+  run $KETCH app deploy "$APP_NAME" --framework "$FRAMEWORK" -i "$APP_IMAGE"
+  [ $status -eq 0 ]
+}
+
+@test "app list" {
+  result=$($KETCH app list)
+  headerRegex="NAME[ \t]+FRAMEWORK[ \t]+STATE[ \t]+ADDRESSES[ \t]+BUILDER[ \t]+DESCRIPTION"
+  dataRegex="$APP_NAME[ \t]+$FRAMEWORK[ \t]+(created|running)"
+  echo "RECEIVED:" $result
+  [[ $result =~ $headerRegex ]]
+  [[ $result =~ $dataRegex ]]
+}
+
+@test "app info" {
+  result=$($KETCH app info "$APP_NAME")
+  headerRegex="DEPLOYMENT VERSION[ \t]+IMAGE[ \t]+PROCESS NAME[ \t]+WEIGHT[ \t]+STATE[ \t]+CMD"
+  dataRegex="1[ \t]+docker.io/shipasoftware/bulletinboard:1.0[ \t]+web[ \t]+100%[ \t]+created[ \t]+docker-entrypoint.sh npm start"
+  echo "RECEIVED:" $result
+  [[ $result =~ $headerRegex ]]
+  [[ $result =~ $dataRegex ]]
+}
+
+@test "app stop" {
+  result=$($KETCH app stop "$APP_NAME")
+  echo "RECEIVED:" $result
+  [[ $result =~ "Successfully stopped!" ]]
+}
+
+@test "app start" {
+  result=$($KETCH app start "$APP_NAME")
+  echo "RECEIVED:" $result
+  [[ $result =~ "Successfully started!" ]]
+}
+
+@test "app log" {
+  run $KETCH app log "$APP_NAME"
+  [ $status -eq 0 ]
```

Comment:
I am not sure how does it work. where did we define this variable?

I generally have seen it is set using:
```
## get status ##
status=$?
```
or

```
#!/bin/bash
/path/to/some/command
[ $? -eq 0 ]  || exit 1
```

Reply:
what will happen if `[ $status -eq 1 ]`? Will it fail?

Reply:
`$status` is the command's exits status. It gets populated by the `BATS` testing framework when using the `run` command specifically for this type of check. `[ $status -eq 1 ]` will fail. I'm _pretty_ sure this is the author's intent: https://bats-core.readthedocs.io/en/stable/writing-tests.html#run-test-other-commands